src/bin/handle-git-dir-change.ts

```diff
@@ -0,0 +1,167 @@
+  fileData: FileData;
+}
+
+// See https://git-scm.com/docs/git-diff#Documentation/git-diff.txt-git-diff-filesltpatterngt82308203
+type GitDiffStatus = 'A' | 'D' | 'M' | 'T' | 'U' | 'X' | string;
+
+/**
+ * Get the absolute path of a relative path
+ * @param {string} dir the wildcard directory containing git change, not necessarily the root git directory
+ * @returns {string} the absolute path relative to the path that the user executed the bash command in
+ */
+export function resolvePath(dir: string) {
+  const absoluteDir = path.resolve(process.cwd(), dir);
+  return absoluteDir;
+}
+
+/**
+ * Get the git root directory.
+ * @param {string} dir the wildcard directory containing git change, not necessarily the root git directory
+ * @returns {string} the absolute path of the git directory root
+ */
+export function findRepoRoot(dir: string): string {
+  try {
+    return execSync('git rev-parse --show-toplevel', {cwd: dir})
+      .toString()
+      .slice(0, -1); // remove the \n
+  } catch (err) {
+    logger.error(`The directory provided is not a git directory: ${dir}`);
+    throw err;
+  }
+}
+
+/**
+ * Get the GitHub mode, file content, and relative path asynchronously
+ * @param {string} gitRootDir the root of the local GitHub repository
+ * @param {string} gitDiffPattern A single file diff. Renames and copies are broken up into separate diffs. See https://git-scm.com/docs/git-diff#Documentation/git-diff.txt-git-diff-filesltpatterngt82308203 for more details
+ * @returns {GitFileData} the current mode, the relative path of the file in the Git Repository, and the file status.
+ */
+export function getGitFileData(
+  gitRootDir: string,
+  gitDiffPattern: string
+): Promise<GitFileData> {
+  return new Promise((resolve, reject) => {
+    try {
+      const fields: string[] = gitDiffPattern.split(' ');
+      const newMode = fields[1] as FileMode;
+      const oldMode = fields[0].substring(1) as FileMode;
+      const statusAndPath = fields[4].split('\t');
+      const status = statusAndPath[0] as GitDiffStatus;
+      const relativePath = statusAndPath[1];
+      // if file is deleted, do not attempt to read it
+      if (status === 'D') {
+        resolve({path: relativePath, fileData: new FileData(null, oldMode)});
+      } else {
+        // else read the file
+        readFile(
+          gitRootDir + '/' + relativePath,
+          {
+            encoding: 'utf-8',
+          },
+          (err, content) => {
+            if (err) {
+              logger.error(
+                `Error loading file ${relativePath} in git directory ${gitRootDir}`
+              );
+              reject(err);
+            }
+            resolve({
+              path: relativePath,
+              fileData: new FileData(content, newMode),
+            });
+          }
+        );
+      }
+    } catch (err) {
+      logger.warning(
+        `\`git diff --raw\` may have changed formats: \n ${gitDiffPattern}`
+      );
+      reject(err);
+    }
```

Comment:
A catch block around a long section of code makes it difficult to see which error is being handled. If this is to catch the parsing error, try to wrap just the parsing with the try block.